core/lib/zksync_core/src/eth_watch/event_processors/governance_upgrades.rs

```diff
@@ -0,0 +1,117 @@
+use crate::eth_watch::{
+    client::{Error, EthClient},
+    event_processors::EventProcessor,
+};
+use std::convert::TryFrom;
+use std::time::Instant;
+use zksync_contracts::governance_contract;
+use zksync_dal::StorageProcessor;
+use zksync_types::{
+    protocol_version::GovernanceOperation, web3::types::Log, Address, ProtocolUpgrade,
+    ProtocolVersionId, H256,
+};
+
+/// Responsible for saving new protocol upgrade proposals to the database.
+#[derive(Debug)]
+pub struct GovernanceUpgradesEventProcessor {
+    diamond_proxy_address: Address,
+    last_seen_version_id: ProtocolVersionId,
+    upgrade_proposal_signature: H256,
+}
+
+impl GovernanceUpgradesEventProcessor {
+    pub fn new(
+        diamond_proxy_address: Address,
+        last_seen_version_id: ProtocolVersionId,
```

Comment:
mention what is 'last_seen_version_id' used for (that we're skipping the upgrades that are smaller than this)